### PR TITLE
8240632: Note differences between IEEE 754-2019 math lib special cases and java.lang.Math

### PR DESCRIPTION
**Pull request by jddarcy**
Recent revisions of the IEEE 754 floating-point standard have added guidance on how typical math library methods (sin, cos, tan, etc.) should behave in terms of their general quality of implementation as well as on special values.

Other than the pow methods, for the recommended operations listed by IEEE 754 that the Java math library already includes, the special cases that are specified by Java are the same as those specified by IEEE 754, except for the pow method. IEEE 754 calls out some special cases not explicitly listed in the Java specs. This changeset adds those special cases to the spec and adds tests of the the special cases if not already present.

If method "Foo" already had a regression test, new cases were added it it. Otherwise, a new test was added to cover the special cases of several methods.

There is no intention at the moment to change the behavior of pow to align with IEEE 754.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8240632](https://bugs.openjdk.java.net/browse/JDK-8240632): Note differences between IEEE 754-2019 math lib special cases and java.lang.Math

### Reviewers
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2395/head:pull/2395`
`$ git checkout pull/2395`

### CODE REVIEW

**Comment by jddarcy** [comment]
PS After the wording for the Math class is finalized, I'll add analogous wording to the StrictMath class and run a CSR.

File: src/java.base/share/classes/java/lang/Math.java [hunk 1]
```diff
@@ -209,7 +233,9 @@ public static double asin(double a) {
      * Returns the arc cosine of a value; the returned angle is in the
      * range 0.0 through <i>pi</i>.  Special case:
      * <ul><li>If the argument is NaN or its absolute value is greater
-     * than 1, then the result is NaN.</ul>
+     * than 1, then the result is NaN.
+     * <li>If the argument is {@code 1.0}, the result is {@code +0.0}.
```
Comment (bplb) [inline src/java.base/share/classes/java/lang/Math.java:237]: `+0.0` is used here but in several cases below  it is written out as `positive zero`.

File: src/java.base/share/classes/java/lang/Math.java [hunk 2]
```diff
@@ -660,6 +705,16 @@ public static double atan2(double y, double x) {
...
      *
+     * @apiNote
+     * The special cases definitions of this method differ from the
+     * special case definitions of the IEEE 754 recommended {@code
+     * pow} operation for &plusmn;{@code 1.0} raised to an infinite
+     * power. This method treats such cases as indeterminate and
+     * specifies a NaN is returned. The IEEE 754 specification treats
+     * the infinite power as a large integer (large-magnitude
+     * floating-point numbers are numerically integers, specifically
+     * even integer) and therefore specifies {@code 1.0} be returned.
```
Comment (bplb) [inline src/java.base/share/classes/java/lang/Math.java:716]: In this sentence `integer` should be plural.

**Review by bplb — APPROVED** [review]

**Pull request closed without merging**